slack_bot.py

```diff
@@ -1064,20 +1087,66 @@ def _parse_list(self, command):
-        On poll method helps to establish the web socket connection  with slack bot app on you slack
+        On poll method helps to establish the web socket connection with slack bot app on you slack
         using the slack-bolt sdk.
         """
         app = slack_app(token=self.bot_token)
 
+        # decorator to catch responses to actions. Using RE to confirm the payload is in the format we expect
+        # it should be given it is a response to an interactive button sent by ask_question
+        @app.action({
+            "type": "interactive_message",
+            "callback_id": re.compile('{"qid": ".*')
+        })
+        def interactive_message_handler(ack, body, respond):
+            """
+            interactive_message_handler is used to handle responses to buttons actions sent by the ask_question
+            action. It receives a json body which contains the data of the event. The contents of the event are
+            written to the $PHANTOM_HOME/local_data/app_stats/$APP_ID folder. The ask_question action runs a loop
+            looking for files in this directory - it will pickup and process the file once it is written.
+            """
+            ack()
+            try:
+                if body:
+                    callback_id = body.get('callback_id')
+                    callback_json = json.loads(UnicodeDammit(callback_id).unicode_markup)
+                    qid = callback_json.get('qid')
+                    confirmation_message = callback_json.get('confirmation')
+                    state_dir = '{0}/{1}'.format(APPS_STATE_PATH, SLACK_APP_ID)
+
+                    answer_filename = '{0}.json'.format(qid)
+                    answer_path = "{0}/{1}".format(state_dir, answer_filename)
+                    tmp_log('**going to put answer file here: {}'.format(answer_path))
+
+                    final_payload = process_payload(body, answer_path)
+
+                    try:
+                        with open(answer_path, 'w') as answer_file:  # nosemgrep
+                            answer_file.write(json.dumps(final_payload))
+                    except Exception as e:
+                        print('Exception occured while writing reponse to {}. Exception: {}'.format(answer_path, e))
+                    respond(confirmation_message)
+            except Exception as e:
+                print('Unknown exception occured while processing answer response. Exception: {}'.format(e))
+
         @app.event("app_mention")
-        def mention_handler(body):
+        def mention_handler(body, say):
             """
             mention handler function uses a app_mention event decorator to response to the events whenever the bot
             is mentioned in the cat. It receives a json body which contains the data of the event. The command and
             channel name are parsed from the body and passed to command handler to further process the command.
             """
+            tmp_log('**app_mention handler hit')
```

Comment:
out of curiosity why did we decide to do log to a temp file instead of doing regular log calls? the downside is this log will never be captured in skynet as we pull in the other SOAR logs

Reply:
As we can observe in the previous builds of  bot script also, the slack bot script never import the phantom base connector class. Therefore we cannot use the regular logging statements we use in connector files. Also the bot script can we used locally after preparing a python config file. So even if we add the base connector, it won't be able to base connector class. 
The **tem_logs**  method is added by the external contributor to log data from the **bot script**.